rfcs/20240425-mha-indirect-kv/README.md

```diff
@@ -0,0 +1,131 @@
+# Introduction
+
+## KV cache
+
+LLMs are auto-regressive models -- each iteration they predict the next token based
+on previous tokens (also called context). The new token is added to the context
+and the model predicts the next token. This process repeats until a stop
+condition is met (for example, max length is exceeded or `<EOS>` symbol is predicted).
+As explained in [[#1]][1] the Multi-Head Attention is the core operation
+that is responsible for the influence of the context on the output token.
+
+Here is the diagram:
+![MHA pattern](mha-pattern.png)
```

Comment:
in this pattern graph, out shape of BMM(N, T) seems be (N, H, T, T)  not (N, H, 1, T) ?